src/cargo/sources/path.rs

```diff
@@ -124,6 +124,9 @@ impl<'gctx> PathSource<'gctx> {
     }
 
     fn read_package(&self) -> CargoResult<Package> {
+        if crate::util::toml::is_embedded(&self.path) {
```

Comment:
I am not sure if this is the best place to check this. Maybe we should move it to where we create the source ID?

Reply:
imo this feels too spooky as this function is not directly tied to dependencies.

Reply:
Then I guess we should do the check when we read the manifest file.

Reply:
Maybe it isn't so bad? If we view the lack of dependency / workspace support as not far away, this just needs to hold us over until then. If its far away or unlikely, then more care might be needed.

Reply:
Dependency support is dependent on either
- `[lib]` support which is hazy
- artifact dependencies which I'd really like to move forward for build script delegation

Workspace support could be really useful for cargo-xtasks

These were deferred out of the first RFC just to focus on the minimal, core feature set.  I am not lined up to immediately follow with these features but I could see them being high demand when this gets stabilized.

Reply:
I have moved it before we create the source ID for the path. I think this location is better than the previous one (read package). It also helped us bail out an error earlier.